run_md.py

```diff
@@ -0,0 +1,879 @@
+        wdir_system_ligand = os.path.join(wdir, 'md_preparation', 'system_lig')
+
+        wdir_md = os.path.join(wdir, 'md_preparation', 'md_files')
+
+        os.makedirs(wdir_md, exist_ok=True)
+        os.makedirs(wdir_protein, exist_ok=True)
+        os.makedirs(wdir_ligand, exist_ok=True)
+        os.makedirs(wdir_system_ligand, exist_ok=True)
+
+        if not os.path.isfile(protein):
+            raise FileExistsError(f'{protein} does not exist')
+
+        pname, p_ext = os.path.splitext(os.path.basename(protein))
+        # check if already exist in the working directory
+        if not os.path.isfile(f'{os.path.join(wdir_protein, pname)}.gro') or not os.path.isfile(os.path.join(wdir_protein, "topol.top")):
+            if p_ext != '.gro' or topol is None or posre_list_protein is None:
+                try:
+                    subprocess.check_output(f'gmx pdb2gmx -f {protein} -o {os.path.join(wdir_protein, pname)}.gro -water tip3p -ignh '
+                          f'-i {os.path.join(wdir_protein, "posre.itp")} '
+                          f'-p {os.path.join(wdir_protein, "topol.top")}'
+                          f'<<< {forcefield_num}', shell=True)
+                except subprocess.CalledProcessError as e:
+                    logging.error(e)
+                    return None
+            else:
+                if not os.path.isfile(os.path.join(wdir_protein, os.path.basename(protein))):
+                    shutil.copy(protein, os.path.join(wdir_protein, os.path.basename(protein)))
+                if not os.path.isfile(os.path.join(wdir_protein, 'topol.top')):
+                    shutil.copy(topol, os.path.join(wdir_protein, 'topol.top'))
+                # multiple chains
+                for posre_protein in posre_list_protein:
+                    if not os.path.isfile(os.path.join(wdir_protein, os.path.basename(posre_protein))):
+                            shutil.copy(posre_protein, os.path.join(wdir_protein, os.path.basename(posre_protein)))
+                if topol_itp_list is not None:
+                    for topol_itp in topol_itp_list:
+                        if not os.path.isfile(os.path.join(wdir_protein, os.path.basename(topol_itp))):
+                            shutil.copy(topol_itp, os.path.join(wdir_protein, os.path.basename(topol_itp)))
+
+
+        else:
+            logging.warning(f'{os.path.join(wdir_protein, pname)}.gro and topol.top files exist. '
+                            f'Protein preparation step will be skipped.')
+
+
+        # Part 1. Preparation. Run on each cpu
+        dask_client = init_dask_cluster(hostfile=hostfile, n_tasks_per_node=ncpu, ncpu=ncpu)
+        try:
+            system_lig_data = [] # wdir_ligand_cur, molid, resid
+            if system_lfile is not None:
+                if not os.path.isfile(system_lfile):
+                    raise FileExistsError(f'{system_lfile} does not exist')
+
+                mols = supply_mols(system_lfile, set_resid=None)
```

Comment:
What happens if file is gro? Probably error?
Should we support gro files? It complicates arguments and the pipeline.